src/containers/App.tsx

```diff
@@ -3,14 +3,32 @@ import Menu from 'components/menu';
 import Viewer from 'components/viewer';
 import Editor from 'components/editor';
 import Highlights from 'components/highlights/Highlights';
+import {connect} from 'react-redux';
 
-export default function App() {
-  return (
-    <div>
-      <Editor />
-      <Highlights />
-      <Menu />
-      <Viewer />
-    </div>
-  );
+interface AppProps {
+  editor: any;
+  viewerVisible: boolean;
+  menuVisible: boolean;
+}
+
+@connect(
+  state => ({
+    editor: state.widgets.editor,
+    viewerVisible: state.widgets.viewer.visible,
+    menuVisible: state.widgets.menu.visible,
+  }),
+)
+export default class App extends React.Component<Partial<AppProps>> {
+  // always updates...:
+  // <Editor key={JSON.stringify(this.props.editor.range) + JSON.stringify(this.props.editor.annotationId)}/>
+  render() {
+    return (
+      <div>
+        {this.props.editor.visible && <Editor/>}
```

Comment:
I'd propose to eventually move Widget management to some kind of `WidgetManager`, which would render appropriate widgets.

Maybe it could also provide some higher-level widget declarations, so we don't have to modify it after each new one is defined (but that's just a general future idea)

Reply:
Would it involve some more widget-related logic than current `App`? I'm not sure I understand the second part. You mean not to manually insert widget components? (like `<Editor>`?)

I think it might be helpful in the future when there are more widgets involved. Right now it's quite handy to be able to manually experiment with them and not "freeze" the widget configuration.

Reply:
I had in mind sth like:

```
@connect(state => ({
  widgetsState: state.widgets,
})
WidgetManager {
  this.widgets = [
    Editor,
    Viewer,
    etc.,
    // add here any new ones?
  ];

  render() {
    const {
      widgetsState,
    } = this.props;

    return (
      <div>
        {this.widgets.map(widget => (
          widgetsState[widget.reducerName].visible && <widget key={widget.name} />
        )}
      </div>
    );
}
```

however, as I said, that's more of a future idea to streamline widget visibility management. As of now it probably is not really required, especially that we don't really plan to add more any more widgets (unless this could help somehow with e.g. 2+ editors being shown at once).


As of now I'd just move the rendering to a dedicated component.